routers/web/user/setting/account.go

```diff
@@ -234,7 +234,11 @@ func DeleteEmail(ctx *context.Context) {
 func DeleteAccount(ctx *context.Context) {
 	ctx.Data["Title"] = ctx.Tr("settings")
 	ctx.Data["PageIsSettingsAccount"] = true
-
+	passwd := ctx.FormString("password")
+	if len(passwd) == 0 || !ctx.Doer.ValidatePassword(passwd) {
```

Comment:
Thanks for making this change. `auth.UserSignIn` below should return `ErrUserPasswordInvalid` if the password is invalid, so perhaps that could be checked in an if conditional instead (pseudocode `err == ErrUserPasswordInvalid`)?

Reply:
I will add this check too. But lil bit confused with the error stack that I have create in above comments are right way of doing it? Need a bit of help there. 
The committed solution will just works as fine. But there are other errors thats may occurs that needs to be covered.